### PR TITLE
[DD4hep] Filtered View is not a Shape

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Shape type check is performed on a FilteredView current shape rather then on a View itself

Yet another small step towards code migration to the DD4hep repo.

@cvuosalo, @namapane - unfortunately, it's bundled with #31578

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
please test

**Comment by namapane** [comment]
@ianna OK but now I don't know how to proceed. The plan was to replace #31578 with a different PR avoiding the replication of code into DD4hep_volumeHandle.cc. I don't know what to do at this point...

**Comment by ianna** [comment]
> @ianna OK but now I don't know how to proceed. The plan was to replace #31578 with a different PR avoiding the replication of code into DD4hep_volumeHandle.cc. I don't know what to do at this point...

@namapane - when your PR is ready, I'll rebase this one.

**Comment by ianna** [comment]
please test with #31605

**Comment by kpedro88** [comment]
+upgrade

File: MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc [hunk 1]
```diff
@@ -197,9 +203,483 @@ namespace {
...
+  // recalculate center: (for a DDCons, DDD gives 0,0,Z)
+  double rZmin = (rInMinusZ + rOutMinusZ) / 2.;
+  double rZmax = (rInPlusZ + rOutPlusZ) / 2.;
+  double rCentr = (rZmin + rZmax) / 2.;
+  Geom::Phi<double> phiCenter(startPhi + deltaPhi / 2.);
+  center_ = refPlane->toGlobal(LocalPoint(rCentr * cos(phiCenter), rCentr * sin(phiCenter), 0.));
+  // For cons and tubs RN = R.
+  theRN = rCentr;
+
+  const double epsilon = 1e-5;
```
Comment (jpata) [inline MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:548]: could also be static constexpr

File: MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc [hunk 2]
```diff
@@ -197,9 +203,483 @@ namespace {
...
+  // Compute angle of the cut plane
+  // Got any easier formula?
+  // FIXME: check that it still holds for cutAtStart > cutAtDelta
+  double c = sqrt(cutAtDelta * cutAtDelta + cutAtStart * cutAtStart - 2 * cutAtDelta * cutAtStart * cos(deltaPhi));
+  double alpha = startPhi - asin(sin(deltaPhi) * cutAtDelta / c);
+  GlobalVector x_Rplane = refPlane->toGlobal(LocalVector(cos(alpha), sin(alpha), 0));
+  Surface::RotationType rot_R(planeZAxis, x_Rplane);
+
+  // FIXME: use builder
+  surfaces[plane_side] = new Plane(pos_Rplane, rot_R);
```
Comment (jpata) [inline MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:659]: for my understanding, where are these objects deleted later? Going through the volumeHandle base classes, I was not immediately able to idenfity it, but I wasn't looking in the right place probably.

**Review by jpata — COMMENTED** [review]
I'm ready to approve it from the reco side, given the limited/minor reco changes in MagneticField/GeomBuilder. I just had a minor comment and a question - please see above.

File: MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc (hunk 2 above)
Comment (namapane) [reply MagneticField/GeomBuilder/src/DD4hep_volumeHandle.cc:659]: Surfaces are reference counted. cf. https://github.com/cms-sw/cmssw/blob/master/DataFormats/GeometrySurface/interface/Surface.h#L36

**Comment by namapane** [comment]
> @namapane - when your PR is ready, I'll rebase this one.

@ianna, you can proceed with rebasing on top of #31649. Thanks.

**Comment by ianna** [comment]
+1

**Comment by ianna** [comment]
replaced by https://github.com/cms-sw/cmssw/pull/31691

**Pull request closed without merging**